src/console/Command.cpp

```diff
@@ -0,0 +1,103 @@
+#include "console/Command.hpp"
+#include "storm/Error.hpp"
+
+// Extern
+TSHashTable<CONSOLECOMMAND, HASHKEY_STRI> g_consoleCommandHash;
+char g_commandHistory[HISTORY_DEPTH][CMD_BUFFER_SIZE];
+uint32_t g_commandHistoryIndex;
+
+
+void ConsoleCommandDestroy() {
+    g_consoleCommandHash.Clear();
+}
+
+char* ConsoleCommandHistory(uint32_t index) {
+    // Return a pointer to the buffer at the specified index
+    return g_commandHistory[((g_commandHistoryIndex + (HISTORY_DEPTH - 1) - index) & (HISTORY_DEPTH - 1))];
+}
+
+uint32_t ConsoleCommandHistoryDepth() {
+    return HISTORY_DEPTH;
+}
+
+void ConsoleCommandInitialize() {
+    ConsoleCommandRegister("help", ConsoleCommand_Help, CONSOLE, "Provides help information about a command.");
+}
+
+int32_t ConsoleCommandRegister(const char* command,
+                               int32_t (*handler)(const char*, const char*),
+                               CATEGORY category,
+                               const char* helpText) {
+    STORM_ASSERT(command);
+    STORM_ASSERT(handler);
+
+    if (SStrLen(command) > (MAX_CMD_LENGTH - 1) || g_consoleCommandHash.Ptr(command)) {
+        // The command name exceeds MAX_CMD_LENGTH, minus the null terminator
+        // or it has already been registered
+        return 0;
+    }
+
+
+    // Register the new command
+    auto commandPtr = g_consoleCommandHash.New(SStrHashHT(command), command, 0, 0);
+
+    commandPtr->command = command;
+    commandPtr->handler = handler;
+    commandPtr->helpText = helpText;
+    commandPtr->category = category;
+
+    return 1;
+}
+
+void ConsoleCommandUnregister(const char* command) {
+    if (command) {
+        auto commandPtr = g_consoleCommandHash.Ptr(command);
+        if (commandPtr) {
+            g_consoleCommandHash.Unlink(commandPtr);
+        }
+    }
+}
+
+uint32_t ValidateFileName(const char* filename);
+
+
+//Commands
+int32_t ConsoleCommand_Help(const char* command, const char* arguments) {
+    // TODO
+    return 0;
+}
+
+int32_t ConsoleCommand_Quit(const char* command, const char* arguments) {
+    // TODO
+    // ConsolePostClose()
+    return 0;
+}
+
+int32_t ConsoleCommand_Ver(const char* command, const char* arguments) {
+    // TODO
+    return 0;
+}
+
+// Private
+uint32_t ValidateFileName(const char* filename) {
```

Comment:
Move private-to-source-file functions to the top of the file even if it means the functions aren't sorted alphabetically.

Reply:
If this looks like it was using `BOOL`, the return type would presumably be `int32_t`, right?

Reply:
> If this looks like it was using `BOOL`, the return type would presumably be `int32_t`, right?

Yes, you're right. Marking as resolved as this will be corrected in the follow-up commit.